tests/test_agents/test_registration.py

```diff
@@ -54,7 +54,7 @@ def slow_down_tests() -> Generator:
     log_messages.collection_complete.value,
     log_messages.request_update.value,
     log_messages.response_update.value,
-    "Entered in the 'finished_registration' round for period 0",
+    "local height == remote height; continuing execution...",
```

Comment:
Ok - the actual problem is this:
DegenerateRound should never be allowed to be part of a composed, non-abstract app. So we should do two things:
1. Change the registration app to end in a non-degenerate round which is simply an "idle" round that is never being left.
2. Add a check in the chaining logic that after chaining no degenerate rounds are present in the final app.

Reply:
1. The registration app needs to end in a degenerate round in order to be chainable. The e2e test checks the registration skill, which is not chained and this is why we end up in a degenerate round when it finishes. 
2. I think we already check for this indirectly because we check for orphans.